UI/BuildDesignatorWnd.cpp

```diff
@@ -241,25 +244,103 @@ namespace {
         if (item.build_type == BT_BUILDING) {
+            const BuildingType* building_type = GetBuildingType(item.name);
+            if (!building_type) {
+                boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd;
+                return browse_wnd;
+            }
+
+            // create title, description, production time and cost
+            title = UserString(item.name);
+            main_text = UserString(building_type->Description());
+            total_cost = building_type->ProductionCost(empire_id, candidate_object_id);
+            production_time = building_type->ProductionTime(empire_id, candidate_object_id);
+
+            main_text += "\n\nProduction cost: " + DoubleToString(total_cost, 3, false);
+            main_text += "\nProduction time: " + boost::lexical_cast<std::string>(production_time);
+
+            // show build conditions
+            std::string EnqueueAndLocationConditionFailedText = EnqueueAndLocationConditionDescription(item.name, candidate_object_id, empire_id, true);
+            if (EnqueueAndLocationConditionFailedText.length() != 0)
+                main_text += "\n\n" + EnqueueAndLocationConditionFailedText;
+
+            // create tooltip
             boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd(new IconTextBrowseWnd(
-                ClientUI::BuildingIcon(item.name), UserString(item.name),
-                EnqueueAndLocationConditionDescription(item.name, candidate_object_id, empire_id)));
+                ClientUI::BuildingIcon(item.name), title, main_text));
             return browse_wnd;
-        } else if (item.build_type == BT_SHIP) {
-            const ShipDesign* ship_design = GetShipDesign(item.design_id);
-            const std::string& name = (ship_design ? ship_design->Name() : UserString("SHIP_DESIGN"));
+        }
+
+        // production item is a ship
+        if (item.build_type == BT_SHIP) {
+            const ShipDesign* design = GetShipDesign(item.design_id);
+            if (!design) {
+                boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd;
+                return browse_wnd;
+            }
+
+            // create title, description, production time and cost, hull type
+            title = design->Name(true);
+            main_text = design->Description(true);
+            total_cost = design->ProductionCost(empire_id, candidate_object_id);
+            production_time = design->ProductionTime(empire_id, candidate_object_id);
+
+            main_text += "\n\nProduction cost: " + DoubleToString(total_cost, 3, false);
+            main_text += "\nProduction time: " + boost::lexical_cast<std::string>(production_time);
+
+            main_text += "\n\nHull: " + UserString(design->Hull());
+
+            // load ship parts, stack ship parts that are used multiple times
+            std::string ship_parts_formatted;
+            std::map<std::string, int> ship_part_names;
+            std::vector<std::string, std::allocator<std::string>> parts = design->Parts();
```

Comment:
Please put a whitespace between the two `>>` at the end of `std::allocator<std::string>>`, without the space I get a compile error on OSX.